### PR TITLE
Add ilverify testing

### PR DESCRIPTION
**Pull request by MichalPetryka**

### CODE REVIEW

File: BetterCallSodium/External/MonoPInvokeCallbackAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft Corporation
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#pragma warning disable IDE0052
```
Comment (DaZombieKiller) [inline BetterCallSodium/External/MonoPInvokeCallbackAttribute.cs]: Is it 100% necessary for the `Type` to be private? Just making it `public` would make sense to me _and_ remove the need to suppress a warning.

File: BetterCallSodium/External/MonoPInvokeCallbackAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+// ReSharper disable All
+using System;
+
+namespace AOT
+{
+    [AttributeUsage(AttributeTargets.Method)]
+    internal class MonoPInvokeCallbackAttribute : Attribute
```
Comment (DaZombieKiller) [inline BetterCallSodium/External/MonoPInvokeCallbackAttribute.cs]: nit: `sealed`

File: BetterCallSodium.Tests/ProcessUtils.cs [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+            };
+
+            process.ErrorDataReceived += (_, args) =>
+            {
+                lock (errorLock)
+                    errorBuilder.AppendLine(args.Data);
+            };
+
+            process.Start();
+
```
Comment (DaZombieKiller) [inline BetterCallSodium.Tests/ProcessUtils.cs]: nit: unnecessary blank line

File: BetterCallSodium.Tests/ProcessUtils.cs [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+            process.BeginOutputReadLine();
+            process.BeginErrorReadLine();
+
```
Comment (DaZombieKiller) [inline BetterCallSodium.Tests/ProcessUtils.cs]: nit: unnecessary blank line

File: BetterCallSodium.Tests/ProcessUtils.cs [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+            process.WaitForExit();
+
+            lock (outputLock)
+                output = outputBuilder.ToString().Trim();
+
+            lock (errorLock)
+                error = errorBuilder.ToString().Trim();
+            return process.ExitCode;
```
Comment (DaZombieKiller) [inline BetterCallSodium.Tests/ProcessUtils.cs]: nit: add a new line here

File: BetterCallSodium.Tests/ProcessUtils.cs [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+                                CreateNoWindow = true,
+                                RedirectStandardOutput = true
+                            })!;
+
+                            using StreamReader standardOutput = process.StandardOutput;
+
+                            int cp = int.Parse(string.Concat(standardOutput.ReadLine()!.Where(char.IsDigit)));
+
+                            if (cp <= 0)
+                                throw new InvalidOperationException();
```
Comment (DaZombieKiller) [inline BetterCallSodium.Tests/ProcessUtils.cs]: nit: add a blank line

File: BetterCallSodium/External/MonoPInvokeCallbackAttribute.cs [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+    {
+        private Type type;
+        public MonoPInvokeCallbackAttribute(Type t) { type = t; }
```
Comment (DaZombieKiller) [inline BetterCallSodium/External/MonoPInvokeCallbackAttribute.cs]: Style-wise, do we want to allow single-line methods to be written like this?

File: BetterCallSodium.Tests/ProcessUtils.cs [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+                        {
+                            // ignore
+                        }
+
+                        try
+                        {
+                            int codePage = (int)GetConsoleCodePage();
+
+                            if (codePage <= 0)
+                                throw new Win32Exception();
```
Comment (DaZombieKiller) [inline BetterCallSodium.Tests/ProcessUtils.cs]: nit: add a blank line

File: BetterCallSodium.Tests/ProcessUtils.cs [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace BetterCallSodium.Tests
+{
+    internal class ProcessUtils
```
Comment (DaZombieKiller) [inline BetterCallSodium.Tests/ProcessUtils.cs]: nit: can be a `static class`

File: .config/dotnet-tools.json [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+  "isRoot": true,
+  "tools": {
+    "dotnet-ilverify": {
+      "version": "6.0.0",
+      "commands": [
+        "ilverify"
+      ]
+    }
+  }
+}
```
Comment (DaZombieKiller) [inline .config/dotnet-tools.json]: nit: missing trailing blank line

File: BetterCallSodium/External/MonoPInvokeCallbackAttribute.cs (hunk 1 above)
Comment (MichalPetryka) [reply BetterCallSodium/External/MonoPInvokeCallbackAttribute.cs]: The code was taken from external sources, I don't want to do many changes to it.

File: BetterCallSodium/External/MonoPInvokeCallbackAttribute.cs (hunk 2 above)
Comment (MichalPetryka) [reply BetterCallSodium/External/MonoPInvokeCallbackAttribute.cs]: As above.

File: BetterCallSodium/External/MonoPInvokeCallbackAttribute.cs (hunk 7 above)
Comment (MichalPetryka) [reply BetterCallSodium/External/MonoPInvokeCallbackAttribute.cs]: As the other reviews said.

**Comment by Jack-Edwards** [comment]
LGTM (I have no idea how to scrutinize this).

**Review by thatbakamono — APPROVED** [review]

**Pull request merged**